src/frontend/src/pages/GanttPage/GanttChartComponents/GanttTaskBar/GanttTaskBarEdit.tsx

```diff
@@ -82,12 +93,46 @@ const GanttTaskBarEdit = ({
     }
   }, [bounds, width]);
 
+  const [showAddWorkPackageModal, setShowAddWorkPackageModal] = useState(false);
+
   return (
     <div
       style={{ position: 'relative', width: '100%', marginTop: 10 }}
       onMouseUp={handleMouseUp}
       onMouseMove={handleMouseMove}
     >
+      <AddWorkPackageModal
+        showModal={showAddWorkPackageModal}
+        handleClose={() => setShowAddWorkPackageModal(false)}
+        addWorkPackage={(workPackage) => {
+          const dup = id + event.workPackages.length + 1;
+          addWorkPackage({
+            id: dup,
+            name: workPackage.name,
+            start: new Date(),
+            end: new Date(Date.now() + 1000 * 60 * 60 * 24 * 7),
+            type: 'task',
+            workPackages: [],
+            projectId: event.id,
+            projectNumber: event.projectNumber,
+            carNumber: event.carNumber,
+            styles: {
+              color: GanttWorkPackageTextColorPipe(workPackage.stage),
+              backgroundColor: GanttWorkPackageStageColorPipe(workPackage.stage, WbsElementStatus.Inactive)
+            }
+          });
+
+          createChange({
+            id,
+            eventId: dup,
+            type: 'create-work-package',
+            name: workPackage.name,
+            stage: workPackage.stage,
+            start: new Date(),
+            end: new Date(Date.now() + 1000 * 60 * 60 * 24 * 7)
+          });
+        }}
```

Comment:
this logic is pretty complex for an inline, if you could move it up and leave a comment about what it's doing from a high level that'd be good